res/controllers/PIONEER-DDJ-ERGO-scripts.js

```diff
@@ -0,0 +1,441 @@
+        loopOut = 1;
+    }
+    //loop was disabled
+    if (control === "loop_enabled" && engine.getParameter(group, "loop_enabled") === 0) {
+        loopIn = 0;
+        loopOut = 0;
+    }
+    const channel = group === "[Channel1]" ? 0 : 1;
+    //if not looping, LED on, else toggle LED on every beat
+    //loop in manipulates blink since loop in blinks while waiting for a loop out click
+    if (loopIn === 0) {
+        midi.sendShortMsg(0x90+channel, 0x10, 0x7F);
+    } else if (control === "beat_active" && value === 1) {
+        midi.sendShortMsg(0x90+channel, 0x10, (1-blink[channel])*0x7F);
+        blink[channel] = 1 - blink[channel];
+    }
+    //if not looping, LED on, else toggle LED on every beat
+    if (loopOut === 0) {
+        midi.sendShortMsg(0x90+channel, 0x11, 0x7F);
+    } else if (control === "beat_active" && value === 1) {
+        midi.sendShortMsg(0x90+channel, 0x11, blink[channel]*0x7F);
+    }
+};
+
+//this blinks the faders on every beat and varying intensity based on fader position
+const volumeLEDsCallback = function(value, group) {
+    const channel = group === "[Channel1]" ? 0 : 1;
+    const volume = engine.getParameter(group, "volume");
+    //message to set intensity + two on/off messages since it requires them with two different midinos
+    midi.sendShortMsg(0xB6, 0x13+2*channel, 0x7F*volume);
+    midi.sendShortMsg(0x9B, 0x74+channel, 0x7F*value);
+    midi.sendShortMsg(0x9B, 0x78+channel, 0x7F*value);
+};
+
+//this triggers an LED animation when the filter knob is turned
+const filterLEDsCallback = function(value, group) {
+    const channel = group === "[QuickEffectRack1_[Channel1]]" ? 0 : 1;
+    midi.sendShortMsg(0xB6, 0x1B+channel, value === 0.5 ? 0x40 : 0x7F*value);
+};
+
+//this triggers an LED animation when a track is loaded into the deck
+const loadingAnimationCallback = function(value, group) {
+    if (engine.getParameter(group, "play") !== 1 && engine.getParameter(group, "track_loaded") === 1) {
+        const channel = group === "[Channel1]" ? 0 : 1;
+        midi.sendShortMsg(0x9B, 0x0C+channel, 0x7F);
+    }
+};
+
+//callback connections
+engine.makeConnection("[Channel1]", "beat_active", pulseModeCallback);
+engine.makeConnection("[Channel2]", "beat_active", pulseModeCallback);
+engine.makeConnection("[Channel1]", "beat_active", loopLEDsCallback);
+engine.makeConnection("[Channel2]", "beat_active", loopLEDsCallback);
+engine.makeConnection("[Channel1]", "loop_in", loopLEDsCallback);
+engine.makeConnection("[Channel2]", "loop_in", loopLEDsCallback);
+engine.makeConnection("[Channel1]", "loop_out", loopLEDsCallback);
+engine.makeConnection("[Channel2]", "loop_out", loopLEDsCallback);
+engine.makeConnection("[Channel1]", "loop_enabled", loopLEDsCallback);
+engine.makeConnection("[Channel2]", "loop_enabled", loopLEDsCallback);
+engine.makeConnection("[Channel1]", "beat_active", volumeLEDsCallback);
+engine.makeConnection("[Channel2]", "beat_active", volumeLEDsCallback);
```

Comment:
```suggestion
engine.makeUnbufferedConnection("[Channel1]", "beat_active", volumeLEDsCallback);
engine.makeUnbufferedConnection("[Channel2]", "beat_active", volumeLEDsCallback);
```
`beat_active` can have a very high update interval in case of fractional loops. Use makeUnbufferedConnection here, to prevent event queue overload.